lib/mpool/src/mpool.c

```diff
@@ -258,7 +258,8 @@ mpool_create(const char *home, const struct mpool_cparams *cparams)
             char   path[PATH_MAX];
             size_t n;
 
-            n = snprintf(path, sizeof(path), "%s/%s", home,
+            n = snprintf(path, sizeof(path), "%s%s%s", home,
+                         home[strlen(home) - 1] == '/' ? "" : "/",
```

Comment:
Presumably you need at least one / at the end?  What if I give you a home with ten slashes at the end?

Reply:
No slash at the end is necessary. Regarding your other point, that could definitely be protected against. I can do that in another PR